File: app/src/main/java/com/v2/desafionubank/RxBus.java
package com.v2.desafionubank;

import rx.Observable;
import rx.subjects.PublishSubject;

/**
 * Created by CaioSChristino on 15/04/18.
 */

public class RxBus {
    public RxBus() {
    }

    private PublishSubject<Object> bus = PublishSubject.create();

    public void send(Object o) {
        bus.onNext(o);
    }

    public Observable<Object> toObservable() {
        return bus;
    }
}

File: app/src/main/java/com/v2/desafionubank/RxBus.kt
package com.v2.desafionubank

import rx.Observable
import rx.subjects.PublishSubject

/**
 * Created by csanchez on 19/04/2018.
 */

class RxBus {
    private val bus = PublishSubject.create<Any>()

    fun send(o: Any) {
        bus.onNext(o)
    }

    fun toObservable(): Observable<Any> {
        return bus
    }
}